muspinsim/hamiltonian.py

```diff
@@ -171,3 +174,56 @@ def integrate_decaying(self, rho0, tau, operators=[]):
         result = np.sum(rho0[None, :, :] * intops[:, :, :], axis=(1, 2))
 
         return result
+
+    def fast_evolve(self, sigma_mu, times, other_dimension):
+        """Compute time evolution of a muon polarisation state using this
+           Hamiltonian
+
+        Computes the evolution of a muon polarisation state under this
+        Hamiltonian and returns a sequence of expectation values.
+
+        The muon polarisation is assumed to be first in the system in this
+        computation.
+
+        Arguments:
+            sigma_mu {ndarray} -- Linear combination of Pauli spin matrices in
+                                  the direction of the muon
+            times {ndarray} -- Times to compute the evolution for, in microseconds
+            other_dimension {int} -- Combined dimension of all non-muons in the
+                                     system
+
+        Returns:
+            [ndarray] -- Expectation values
+
+        Raises:
+            ValueError -- Invalid values of times
+            RuntimeError -- Hamiltonian is not hermitian
+        """
+
+        times = np.array(times)
+
+        if len(times.shape) != 1:
+            raise ValueError("times must be an array of values in microseconds")
```

Comment:
Yes I didn't think inheritance was suitable here since Hamiltonian and Hermitian contain methods that wont work with CelioHamiltonian mostly due to the need of adding interaction terms and not the matrices together. I have added a validation.py instead for these cases (utils.py causes a circular import).